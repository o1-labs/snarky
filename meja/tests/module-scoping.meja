module A = {
  type t;
};

module B = {
  type t;

  module A = {
    type t;
  };

  module B = {
    type t;
  };
};

type t;

let f = fun (f, x : t, y : A.t, z : B.t) => {
  f(x, y, z);
};

module C = {
  open B;

  let f = fun (f, x : t, y : A.t, z : B.t) => {
    f(x, y, z);
  };
};

let g = fun (z, x, y) => {
  let x = f(z, x, y);
  let y = C.f(z, x, y);
  (x, y);
};
