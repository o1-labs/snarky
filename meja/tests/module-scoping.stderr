File "tests/module-scoping.meja", line 33, characters 14-15:
Error: Incompatible types B.t -> B.A.t -> B.B.t -> 'a and
       t -> A.t -> B.t -> 'b:
       Cannot unify B.t and t
